src/contracts/proxies/ODSafeManager.sol

```diff
@@ -169,11 +173,17 @@ contract ODSafeManager is IODSafeManager {
   function modifySAFECollateralization(
     uint256 _safe,
     int256 _deltaCollateral,
-    int256 _deltaDebt
+    int256 _deltaDebt,
+    bool _nonSafeHandlerAddress
   ) external safeAllowed(_safe) {
     SAFEData memory _sData = _safeData[_safe];
+    if (_deltaDebt != 0) {
+      ITaxCollector(taxCollector).taxSingle(_sData.collateralType);
+    }
+    address collateralSource = _nonSafeHandlerAddress ? msg.sender : _sData.safeHandler;
+    address debtDestination = collateralSource;
     ISAFEEngine(safeEngine).modifySAFECollateralization(
-      _sData.collateralType, _sData.safeHandler, _sData.safeHandler, _sData.safeHandler, _deltaCollateral, _deltaDebt
+      _sData.collateralType, _sData.safeHandler, collateralSource, debtDestination, _deltaCollateral, _deltaDebt
```

Comment:
Is there any danger with allowing the user to select the incorrect value for `_nonSafeHandlerAddress`?

Maybe lets add some testing (in another issue) to test these edge cases.

Reply:
There is no danger for the protocol, but it's possible the transaction will simply revert.